### PR TITLE
postgres: remove constructors in favor of URLOpeners

### PR DESCRIPTION
**Pull request by zombiezen**
Fixed consistency issues as I found them.

Updates #1175

### CODE REVIEW

**Comment by zombiezen** [comment]
@shantuo, I'd like you to review since @vangent is out next week. I'll have follow-up PRs to do the same for MySQL, but I'm going to take that one slower since it affects guestbook.

I have manually tested each of the packages as I made changes.

File: postgres/cloudpostgres/cloudpostgres.go [hunk 1]
```diff
@@ -100,100 +100,50 @@ func (uo *URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB,
...
 	}
-	params.TraceOpts = uo.TraceOpts
-	db, _, err := Open(ctx, uo.CertSource, params)
-	return db, err
+
+	query := u.Query()
+	for k := range query {
+		// Only permit parameters that do not conflict with other behavior.
+		if k == "sslmode" || k == "sslcert" || k == "sslkey" || k == "sslrootcert" {
+			return nil, fmt.Errorf("cloudpostgres: open: extra parameter %s not allowed; use Params fields instead", k)
```
Comment (vangent) [inline postgres/cloudpostgres/cloudpostgres.go]: "use Params fields instead" is stale.

File: postgres/rdspostgres/rdspostgres.go [hunk 2]
```diff
@@ -33,31 +32,12 @@ import (
...
-	// TraceOpts contains options for OpenCensus.
-	TraceOpts []ocsql.TraceOption
-}
-
 // URLOpener opens RDS PostgreSQL URLs
 // like "rdspostgres://user:password@myinstance.borkxyzzy.us-west-1.rds.amazonaws.com:5432/mydb".
 type URLOpener struct {
+	// CertSource specifies how the opener will obtain the RDS Certificate
+	// Authority. If nil, it will use the default *rds.CertFetcher.
+	CertSource rds.CertPoolProvider
```
Comment (vangent) [inline postgres/rdspostgres/rdspostgres.go]: Why not use the "lazy" pattern used elsewhere here? Is getting the default CertFetcher just super cheap for some reason, while for `cloudpostgres` it is not?

File: postgres/rdspostgres/rdspostgres_test.go [hunk 3]
```diff
@@ -104,6 +61,7 @@ func TestOpenWithURL(t *testing.T) {
 		{fmt.Sprintf("rdspostgres://%s:%s@%s/wrongdbname", username, password, endpoint), false, true},
 		{fmt.Sprintf("rdspostgres://%s:%s@%s/%s?foo=bar", username, password, endpoint, databaseName), false, true},
```
Comment (vangent) [inline postgres/rdspostgres/rdspostgres_test.go]: It looks like you deleted all of the tests verifying that `sslmode` etc. are rejected, can you add some back here?

Comment (zombiezen) [reply postgres/rdspostgres/rdspostgres_test.go]: They are already covered by other test cases, see ~10 lines above.

File: postgres/cloudpostgres/cloudpostgres.go (hunk 1 above)
Comment (zombiezen) [reply postgres/cloudpostgres/cloudpostgres.go]: Done.

File: postgres/rdspostgres/rdspostgres.go (hunk 2 above)
Comment (zombiezen) [reply postgres/rdspostgres/rdspostgres.go]: Yes. `rds.CertFetcher` does not require credentials (which is expensive to fetch), it purely fetches a root CA and can do that on-demand.

File: postgres/rdspostgres/rdspostgres_test.go (hunk 3 above)
Comment (vangent) [reply postgres/rdspostgres/rdspostgres_test.go]: Oh sorry, GitHub being clever and hiding lines.

**Pull request merged**